cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/internal/AliasAnalysis.qll

```diff
@@ -285,16 +282,8 @@ predicate resultPointsTo(Instruction instr, IRVariable var, IntValue bitOffset)
     (
       operandIsPropagated(operand, propagatedBitOffset)
       or
-      exists(CallInstruction ci, FunctionIR f, Instruction init |
-        ci = operand.getUseInstruction() and
-        f.getFunction() = ci.getStaticCallTarget() and
-        (
-          init.(InitializeParameterInstruction).getParameter() = f.getFunction().getParameter(operand.(PositionalArgumentOperand).getIndex())
-          or
-          init instanceof InitializeThisInstruction and
-          init.getEnclosingFunctionIR() = f and
-          operand instanceof ThisArgumentOperand
-        ) and
+      exists(CallInstruction ci, Instruction init |
+        isArgumentForParameter(ci, operand, init) and
```

Comment:
The change to use `isArgumentForParameter` here doesn't just remove code duplication but also changes what the code does. It means that we add `not f.isVirtual()` and `not f instanceof AliasFunction`. That's also what we want, right?

Reply:
Yes, it is